sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -212,6 +212,25 @@ abstract class LogicalPlan extends QueryPlan[LogicalPlan] with Logging {
     }
   }
 
+  /**
+   * Resolves the given `name` string against any known virtual columns, returning either
+   * 0 or 1 match
+   *
+   * Every supported virtual column needs to add a check here so that the logical plan can
+   * resolve it.
+   */
+  private def resolveAsVirtualColumn(
+      nameParts: Seq[String],
+      resolver: Resolver,
+      attribute: Attribute): Option[(Attribute, List[String])] = {
+    assert(nameParts.nonEmpty)
+    if (resolver(nameParts.head, VirtualColumn.sparkPartitionId)) {
+      Option((UnresolvedAttribute(VirtualColumn.sparkPartitionId), Nil))
```

Comment:
OK, you put a `UnresolvedAttribute` at here because `SparkPartitionID` is not in the catalyst subproject. But, the problem is that we are not doing anything useful at here because our parser will create an "UnresolvedAttribute("spark__partition__id" :: Nil)".